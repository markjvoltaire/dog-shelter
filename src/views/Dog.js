import React, { useEffect, useState } from 'react';
import { fetchDogById } from '../services/dog';
import { DogDetail } from '../components/DogDetail';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';

export default function Dog() {
  const [dog, setDog] = useState([]);

  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(params.id);
      setDog(data);
    };
    fetchData();
  }, [params.id]);

  return (
    <div>
      <DogDetail dog={dog} />
    </div>
  );
}
