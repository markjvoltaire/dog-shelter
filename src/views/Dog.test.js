import { render, screen } from '@testing-library/react';
import Dog from './Dog';
import { MemoryRouter, Route } from 'react-router-dom';

test('renders each dog', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/10']}>
      <Route path="/dogs/:id" component={Dog} />
    </MemoryRouter>
  );
  await screen.findByText('My Name is Barton');

  expect(container).toMatchSnapshot();
});
