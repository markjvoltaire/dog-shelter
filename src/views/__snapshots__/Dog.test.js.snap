// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders each dog 1`] = `
<div>
  <div>
    <div>
      <h1>
        My Name is 
        Barton
      </h1>
      <img
        src="https://placedog.net/500?id=6"
      />
      <h2>
        Get To Know Me:
      </h2>
      <p>
        Adores every person he meets. Very social and lots of energy
      </p>
    </div>
  </div>
</div>
`;
